python/sgl_jax/srt/models/grok.py

```diff
@@ -665,11 +677,13 @@ def __call__(
             )
             layers_kv_fused.append(kv_fused)
         # Apply final normalization (matching PyTorch fused_dual_residual_rmsnorm)
+        assert deferred_norm is not None
+        assert residual is not None
         hidden_states, _ = dual_rmsnorm_forward(
             hidden_states,
             residual,
-            deferred_norm.scale,
-            self.norm.scale,
+            cast(jax.Array, deferred_norm.scale),
+            cast(jax.Array, self.norm.scale),
```

Comment:
Can you find a way to limit the type of scale to avoid cast. Or use assert to force it not to be None